internal/ledgerutil/compare.go

```diff
@@ -39,8 +44,19 @@ func Compare(snapshotDir1 string, snapshotDir2 string, outputFile string) (count
 		return 0, errors.New("both snapshots public state hashes are same. Aborting compare")
 	}
 
+	allDivsFilepath := filepath.Join(outputDir, AllDivsFilename)
+	outputFullpath, err := filepath.Abs(outputDir)
+	if err != nil {
+		return 0, err
+	}
+
+	// Check for existing output files
+	if _, err := os.Stat(allDivsFilepath); !os.IsNotExist(err) {
+		return 0, errors.Errorf("%s already exists in %s. Aborting compare", AllDivsFilename, outputFullpath)
+	}
```

Comment:
A better check would be just to test that the dir is not empty. Going forward, you would produce more files and  mixing of files from different run would not be desired.

Reply:
I'm not sure what you mean. Since the user can choose to produce the files in any directory, are you saying the command should fail if the directory they have given the tool is not empty? If that's the case, then shouldn't we also create this directory for the user?

Reply:
Also, files would not be mixed as long as we check for all output files and throw an error if any of them exist.

Reply:
Why would you want to do the work for first file and then return error if the second file is present. What's the point of allowing user to write to a dir that may have some unrelated files. 

By mixing what I meant was that as this tool is still evolving, going forward some of the files are going to be optional. For example, when you would add the files for the private data hashes comparison. If your code is not going to produce these files, likely, you will miss checking for the present of those as well and some files from previous runs may get mixed up. Not sure what is the issue in expecting an empty dir to produce the output. If you want to create a new dir, I am fine with that as well if we can come up with some unique name (e.g., append the two input snapshot folders name).
 What I did not find maintainable was testing for file by file one at a time while producing that file.

Reply:
I think concatenating the two snapshot directory names to create the result directory name should work.